src/parcels/_core/index_search.py

```diff
@@ -44,7 +44,7 @@ def _search_1d_array(
     # TODO v4: We probably rework this to deal with 0D arrays before this point (as we already know field dimensionality)
     if len(arr) < 2:
         return np.zeros(shape=x.shape, dtype=np.int32), np.zeros_like(x)
-    index = np.searchsorted(arr, x, side="right") - 1
+    index = np.clip(np.searchsorted(arr, x, side="right") - 1, 0, len(arr) - 2)
```

Comment:
I'm not 100% sure why this clipping is introduced. Is this related to out of bound searches?